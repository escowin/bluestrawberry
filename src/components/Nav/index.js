import React, { useEffect } from "react";
import { capitalizeFirstLetter } from "../../utils/helpers";

function Nav(props) {
  // defining the passed down props from parent App component
  const { categories = [], setCurrentCategory, currentCategory } = props;

  // hook | first argument = callback function; second argument directs hook to re-render component on changes to this state's value.
  useEffect(() => {
    document.title = capitalizeFirstLetter(currentCategory.name);
  }, [currentCategory]);

  // function categorySelected(name) {console.log(`${name} clicked`)}

  return (
    <header>
      <h1>
        <a data-testid="link" href="/" aria-label="title">
          {" "}
          Bluestrawberry
        </a>{" "}
      </h1>
      <nav>
        <ul className="flex" id="nav-links">
          <li>
            <a data-testid="about" href="#about">
              About
            </a>
          </li>

          {/* dynamically generates remaining li */}
          {categories.map((category) => (
            <li
              className={`series-link ${
                // evaluates ===, if true, navActive is returned
                currentCategory.name === category.name && "navActive"
              }`}
              key={category.name} // map(), outermost element must have unique key attribute
            >
              <span
                onClick={() => setCurrentCategory(category)} // anon arrow function: onClick expects callback function declaration
              >
                {capitalizeFirstLetter(category.name)}
              </span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

export default Nav;
