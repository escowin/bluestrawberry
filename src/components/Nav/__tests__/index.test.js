import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

const categories = [{ name: "volume-one", description: "original run" }];
const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();

describe("Nav compnent", () => {
  // baseline test
  it("renders", () => {
    render(
      <Nav
        categories={categories}
        setCurrentCategory={mockSetCurrentCategory}
        currentCategory={mockCurrentCategory}
      />
    );
  });

  // snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Nav
        categories={categories}
        setCurrentCategory={mockSetCurrentCategory}
        currentCategory={mockCurrentCategory}
      />
    );
    // assert value comparison
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("a element is visible", () => {
  it("is nested within h2", () => {
    // arrange
    const { getByLabelText } = render(
      <Nav
        categories={categories}
        setCurrentCategory={mockSetCurrentCategory}
        currentCategory={mockCurrentCategory}
      />
    );
    // assert
    expect(getByLabelText("title")).toHaveTextContent("Bluestrawberry");
  });
});

describe("nav links are visible", () => {
  it("inserts text into nav links", () => {
    // arrange
    const { getByTestId } = render(
      <Nav
        categories={categories}
        setCurrentCategory={mockSetCurrentCategory}
        currentCategory={mockCurrentCategory}
      />
    );
    // expect
    expect(getByTestId("link")).toHaveTextContent("Bluestrawberry");
    expect(getByTestId("about")).toHaveTextContent("About");
  });
});

afterEach(cleanup);
