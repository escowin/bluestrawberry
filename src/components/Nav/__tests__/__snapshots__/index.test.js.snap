// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav compnent matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        aria-label="title"
        data-testid="link"
        href="/"
      >
         Bluestrawberry
      </a>
       
    </h2>
    <nav>
      <ul
        class="flex"
        id="nav-links"
      >
        <li>
          <a
            data-testid="about"
            href="#about"
          >
            About
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="series-link false"
        >
          <span>
            Volume-one
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
